ASWF/meetings/working_groups/configs/2020-11-03.md

```diff
@@ -0,0 +1,147 @@
+      Alex was at last meeting. We should reach out to him. Since we're pulling 
+      naming from CTL, ideally correct in aces-dev. CTL in flux - some PRs that 
+      when merged will remove a lot of complexity. We can influence change 
+      there similarly to how ACES influences OCIO change.
+    - Michael: Last meeting we discussed that the input BuiltinTransforms were 
+      largely present, but needed output transforms to proceed. Is there an 
+      expected timeline for that work Doug?
+    - Doug: Will try to get something before next meeting.
+    - Mark: OCIO v2 has display transforms. Are we making use of that? Some 
+      transforms have OCES space, but PQ goes straight from scene linear. This 
+      config is supposed to represent all the features of OCIO v2. should we 
+      make use of both?
+    - Doug: Yes, will include this in proposal, but think will make use of 
+      display color spaces, so output transforms will be view transform plus 
+      display transform. OCES wouldn't have an interface. More recent ACES 
+      transforms have moved away from it. OCES was interesting theoretical 
+      idea, providing a break point between points of transform that proved to 
+      be more confusing. Would not use OCES.
+    - Mark: What would become the display space? The generator draws 
+      connections between the ACES blobs and the newer transforms skip the OCES 
+      step.
+    - Doug: Display telemetry. New transforms go through that but don't break 
+      it out as external interface in CTL, for historical reasons.
+    - Mark: That will make it difficult to make generator which looks at CTL to 
+      generate. Not married to that plan but seems sensible for tracking 
+      upstream. Also happy to abandon it for config expressed in code if that's 
+      more useful.
+    - Doug: Was thinking could use generator to identify the ACES end and 
+      display code value end, but not necessarily use the OCES interface. For 
+      output transforms, some go direct from ACES and others through OCES, but 
+      we could handle them as if they went from ACES to device.
+    - Mark: At moment, generator doesn't parse CTL code, just looks at URN in 
+      header of files, so no way to track transforms or chains between functions.
+    - Doug: We know anything that expects OCES as input wants to include RRT.
+    - Mark: So use RRT as point?
+    - Doug: Part of problem is the RRT does not produce display referred 
+      imagery, which has caused confusion. Produces idealized output referred 
+      intermediate. Actual display referred colorimetry requires first part of 
+      ODT. RRT doesn't provide technical benefit. Think we can ignore RRT, and 
+      for transforms with OCES input, go from ACES to OCES.
+    - Mark: Not sure benefit to tracking CTL to get these relationships if we 
+      need to implement transforms. Maybe skip checking of CTL and 
+      programmatically build transforms.
```

Comment:
Hang-on, we are talking about the ACES Reference config, CTL is still, AFAIK, the reference implementation, why not building against it? It has the benefits of:

- Producing a config that matches exactly the reference implementation
- Facilitates the Logo Program
- Surfaces any errors back promptly because it is automated
- The Academy can run it easily to check themselves!

Reply:
There was, as you see, discussion around whether the actual config generation should come from code with more manual intervention, to allow a human touch in designing _how_ the reference implementation utilizes OCIO features. Particularly for the output transform side of things there were some expressed concerns that being strictly upstream automated may be problematic. Even in that case though there would be clear value in using the upstream graphing code to validate what is designed, gathering the named transforms which define ACES and trying to run them with image comparison against the config. Please do discuss though, @doug-walker , @Shootfast and others. I can always capture the discussion from this PR and append it to the notes for a clearer record if that's beneficial. I'm hesitant to encourage having this discussion in Slack since it will vanish eventually (though admittedly I was suggesting that in these notes...). We could also wait until the next meeting when @doug-walker has an example of the output transforms.

Reply:
Just to be clear nothing says everything has to be a strict translation of the CTL structure. I think there is a virtue to it because we get a faithful representation of what is happening upstream and this is what developers tend to implement when they implement ACES on their end. The benefit should be obvious for anyone who has implemented ACES in another language, e.g. HLSL, GLSL: it requires much less work to maintain parity when changes occur in *aces-dev*. Whether you like or not the design, this is the reference implementation and straying out of it too much might cause more harm than good from a maintenance standpoint.

Something worth mentioning, that might have been overlooked and forgotten (because not a lot of people actually used the code) is that the current ACES config is partially generated by introspecting CTL, it does less than the current repo but it was doing most of that and THEN, on top of the discovered stuff, it was adding the extra elements, manually. Experience as shown that it was invaluable to maintain the config and see what changed between ACES releases, especially between 1.0.3 and 1.1.

In any case, if we want to diverge from the upstream structure for a "good" reason, I'm not sure what is the concern from an implementation standpoint. We add a case for it and call it a day... AFAIK, the current code has been written by human intervention, so again what is the concern? :)

> I'm hesitant to encourage having this discussion in Slack since it will vanish eventually (though admittedly I was suggesting that in these notes...).

Slack is the place where information vanishes into the Void thus here seems like a good place to me!

Reply:
Agree that there is value in being able to introspect the aces-dev repo CTL and pull the transform IDs, particularly as new ACES versions come out.  Thomas clearly has a lot of experience in doing these updates and knows how much work is involved and the value of automated tools.